app/javascript/components/App.js

```diff
@@ -1,45 +1,39 @@
-import React from "react"
-import DrinkShow from "./pages/DrinkShow"
-import Header from "./components/Header"
-import Footer from "./components/Footer"
-import NotFound from "./pages/NotFound"
-import Home from "./pages/Home"
-import AboutUs from "./pages/AboutUs"
-import Favorites from "./pages/Favorites"
-import {
-  BrowserRouter as Router,
-  Route,
-  Switch
-} from 'react-router-dom'
+import React from "react";
+import DrinkShow from "./pages/DrinkShow";
+import Header from "./components/Header";
+import Footer from "./components/Footer";
+import NotFound from "./pages/NotFound";
+import Home from "./pages/Home";
+import AboutUs from "./pages/AboutUs";
+import Favorites from "./pages/Favorites";
+import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
 
 class App extends React.Component {
-  constructor(props){
-    super(props)
+  constructor(props) {
+    super(props);
     this.state = {
       drinks: [],
-      redirectToReferrer:false
-
-    }
+      redirect: false
```

Comment:
as discussed before, consider moving the postToFavorites() call into the component that really cares about that request, so that you can also move this redirect into that component